internal/controller/nonadminbackup_controller.go

```diff
@@ -69,44 +72,98 @@ const (
 func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	r.Log = log.FromContext(ctx)
 	logger := r.Log.WithValues("NonAdminBackup", req.NamespacedName)
+	logger.V(1).Info(">>> Reconcile NonAdminBackup - loop start")
 
+	// Resource version
+	// Generation version - metadata - only one is updated...
 	r.Context = ctx
 	r.NamespacedName = req.NamespacedName
 
+	// Get the Non Admin Backup object
 	nab := nacv1alpha1.NonAdminBackup{}
 	err := r.Get(ctx, req.NamespacedName, &nab)
 
-	if err != nil && errors.IsNotFound(err) {
-		logger.V(1).Info("Deleted NonAdminBackup CR", nameField, req.Name, "Namespace", req.Namespace)
+	// Bail out when the Non Admin Backup reconcile was triggered, when the NAB got deleted
+	// Reconcile loop was triggered when Velero Backup object got updated and NAB isn't there
+	if err != nil && apierrors.IsNotFound(err) {
+		logger.V(1).Info("Deleted NonAdminBackup CR", nameField, req.Name, constant.NameSpaceString, req.Namespace)
 		return ctrl.Result{}, nil
 	}
 
 	if err != nil {
-		logger.Error(err, "Unable to fetch NonAdminBackup CR", nameField, req.Name, "Namespace", req.Namespace)
+		logger.Error(err, "Unable to fetch NonAdminBackup CR", nameField, req.Name, constant.NameSpaceString, req.Namespace)
 		return ctrl.Result{}, err
 	}
 
-	backupSpec, err := function.GetBackupSpecFromNonAdminBackup(&nab)
-
-	if backupSpec == nil {
-		logger.Error(err, "NonAdminBackup CR does not contain valid BackupSpec")
-		return ctrl.Result{}, nil
+	if nab.Status.Phase == constant.EmptyString {
+		// Phase: New
+		updatedStatus, errUpdate := function.UpdateNonAdminPhase(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminBackupPhaseNew)
+		if updatedStatus {
+			logger.V(1).Info("NonAdminBackup CR - Rqueue after Phase Update")
+			return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, nil
+		}
+		if errUpdate != nil {
+			logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: New", nameField, req.Name, constant.NameSpaceString, req.Namespace)
+			return ctrl.Result{}, errUpdate
+		}
 	}
 
+	backupSpec, err := function.GetBackupSpecFromNonAdminBackup(&nab)
+
 	if err != nil {
-		logger.Error(err, "Error while performing NonAdminBackup reconcile")
+		errMsg := "NonAdminBackup CR does not contain valid BackupSpec"
```

Comment:
This main tasks of this whole code block seems like:
- Set `Phase: BackingOff`
- Set Condition `BackupAccepted: False`
- Set Condition `BackupQueued: False` (seems to be missing )
So why not combine these tasks into one function like `SetPhaseAndCondition` ? This will make the main reconciler more readable and more debug-gable ? WDYT ?